src/relay/backend/build_module.cc

```diff
@@ -437,28 +441,50 @@ class RelayBuildModule : public runtime::ModuleNode {
     ret_.params = graph_codegen_->GetParams();
 
     auto lowered_funcs = graph_codegen_->GetLoweredFunc();
+
+    // When there is no lowered_funcs due to reasons such as optimization,
+    // we first try to generate a dummy one if the target host is "llvm".
+    if (lowered_funcs.size() == 0) {
+      // Decide first the target host 
+      Target target_host_val = target_host_;
+      if (!target_host_.defined()) {
+        for (const auto &it: targets_) {
+          if (it.second->device_type == kDLCPU) {
+            target_host_val = it.second;
+            break;
+          }
+        }
+      }
+
+      // If no target_host has been set, we choose a default one, which is
+      // llvm if "codegen.build_llvm" is accessible.
+      const runtime::PackedFunc* pf = runtime::Registry::Get("codegen.build_llvm");
+      if (!target_host_val.defined())
+        target_host_val = (pf != nullptr) ? target::llvm() : target::stackvm();
+
+      if (target_host_val.defined() && target_host_val->str() == "llvm")
```

Comment:
Please add one test "llvm -device=arm_cpu -mattr=+neon -target=aarch64-linux-gnu" to make sure we get correct llvm target. I doubt your `target_host_val->str() == "llvm"` is not correct logic.

Reply:
Thanks @FrozenGene, this is indeed problematic. I changed the condition to `target_host_val->target_name == "llvm"`. Hope that's OK.